test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java

```diff
@@ -81,11 +81,8 @@ public boolean run() throws Exception {
             byte[] jpegData;
             BufferedImage thumbnail;
             try (ByteArrayOutputStream byteOut = new ByteArrayOutputStream()) {
-                try {
-                    thumbnail = writeImage(byteOut);
-                } finally {
-                    jpegData = byteOut.toByteArray();
-                }
+                thumbnail = writeImage(byteOut);
+                jpegData = byteOut.toByteArray();
             }
```

Comment:
For `ByteArrayOutputStream`, it should be okay to convert it before the stream is closed, and the stream is closed by `writeImage` anyway. However, placing `jpegData = byteOut.toByteArray()` after the try-with-resources for `byteOut` won't raise any questions. Then, `jpegData` can be declared where it's assigned.

Reply:
If I understood you correctly:
I couldn't place `jpegData = byteOut.toByteArray()` after the try-with-resources block because `byteOut` fell out of scope.

But if the priority is to declare variables where they're assigned: I can just remove that inner try-with-resources block. (This is updated.)

Reply:
I missed that `byteOut` was declared inside the try-block only. More in [the following comment](https://github.com/openjdk/jdk/pull/23920#discussion_r2036131572), which I should've posted as a reply to this thread.